src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/ScreenshotInProcess.cs

```diff
@@ -0,0 +1,593 @@
+    {
+        Grayscale = 0,
+        TrueColor = 2,
+        Indexed = 3,
+        GrayscaleWithAlpha = 4,
+        TrueColorWithAlpha = 6,
+    }
+
+    private enum PngCompressionMethod : byte
+    {
+        /// <summary>
+        /// <see href="https://www.w3.org/TR/png/#dfn-deflate">deflate</see> compression with a sliding window of at
+        /// most 32768 bytes.
+        /// </summary>
+        Deflate = 0,
+    }
+
+    private enum PngFilterMethod : byte
+    {
+        /// <summary>
+        /// Adaptive filtering with five basic filter types.
+        /// </summary>
+        Adaptive = 0,
+    }
+
+    private enum PngInterlaceMethod : byte
+    {
+        /// <summary>
+        /// No interlacing.
+        /// </summary>
+        None = 0,
+
+        /// <summary>
+        /// Adam7 interlace.
+        /// </summary>
+        Adam7 = 1,
+    }
+
+    private enum ApngDisposeOp : byte
+    {
+        /// <summary>
+        /// No disposal is done on this frame before rendering the next; the contents of the output buffer are left
+        /// as-is.
+        /// </summary>
+        None = 0,
+
+        /// <summary>
+        /// The frame's region of the output buffer is to be cleared to fully transparent black before rendering the
+        /// next frame.
+        /// </summary>
+        Background = 1,
+
+        /// <summary>
+        /// The frame's region of the output buffer is to be reverted to the previous contents before rendering the next
+        /// frame.
+        /// </summary>
+        Previous = 2,
+    }
+
+    private enum ApngBlendOp : byte
+    {
+        /// <summary>
+        /// All color components of the frame, including alpha, overwrite the current contents of the frame's output
+        /// buffer region.
+        /// </summary>
+        Source = 0,
+
+        /// <summary>
+        /// The frame should be composited onto the output buffer based on its alpha, using a simple OVER operation as
+        /// described in <see href="https://www.w3.org/TR/png/#13Alpha-channel-processing">Alpha Channel Processing</see>.
+        /// </summary>
+        Over = 1,
+    }
+
+    static ScreenshotInProcess()
+    {
+        DataCollectionService.RegisterCustomLogger(
+            static fullPath =>
+            {
+                lock (s_frames)
+                {
+                    if (s_frames.Count == 0)
+                        return;
+                }
+
+                // Try to capture an additional frame at the end
+                s_currentInstance?.CaptureFrame();
+
+                (TimeSpan elapsed, BitmapSource image)[] frames;
+                lock (s_frames)
+                {
+                    if (s_frames.Count < 2)
+                    {
+                        // No animation available
+                        return;
+                    }
+
+                    frames = s_frames.ToArray();
+                }
+
+                Array.Sort(frames, (x, y) => x.elapsed.CompareTo(y.elapsed));
```

Comment:
➡️ Comment added. It's a bit clumsy but gets the job done. I expect it will be improved if we decide to productize this functionality across repositories.